src/main/java/org/jenkins_ci/plugins/flexible_publish/FlexiblePublisher.java

```diff
@@ -112,9 +113,17 @@ public boolean prebuild(final AbstractBuild<?, ?> build, final BuildListener lis
     @Override
     public boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher, final BuildListener listener)
                                                                                                 throws InterruptedException, IOException {
-        for (ConditionalPublisher publisher : publishers)
-            if (!publisher.perform(build, launcher, listener))
+        for (ConditionalPublisher publisher : publishers) {
+            try {
+                if (!publisher.perform(build, launcher, listener)) {
+                    setResult(build, Result.FAILURE);
+                    listener.getLogger().println("[flexible-publish] Publisher "+publisher.getDescriptor().getDisplayName() + "failed build");
+                }
+            } catch (AbortException ex){
                 setResult(build, Result.FAILURE);
+                listener.getLogger().println("[flexible-publish] Publisher "+publisher.getDescriptor().getDisplayName() + "failed build");
```

Comment:
I'm using this plugin with this patch and realized that we logging must be enhanced
1) ex.getMessage() - contains the reason of build failure if publisher stopped working with 'return false'
2) `publisher.getDescriptor().getDisplayName()` is wrong, because it return description of conditional step
3) It good for prefixing publisher message with publisher name. For this plugin i expect the next
`[flexible-publish] Conditional step XXXX failed with error ex.getMessage()` if it was thrown with Exception
and print `[flexible-publish] Conditional step XXXX failed` assuming that publisher already printed error message to listener before returning boolean